c_api/blockingconcurrentqueue.cpp

```diff
@@ -0,0 +1,48 @@
+#include "concurrentqueue.h"
+#include "../blockingconcurrentqueue.h"
+
+typedef moodycamel::BlockingConcurrentQueue<void*> MoodycamelBCQType, *MoodycamelBCQPtr;
+
+extern "C" {
+	
+int moodycamel_bcq_create(MoodycamelBCQHandle* handle)
+{
+	if (handle == nullptr) {
+		return 0;
+	}
+	MoodycamelBCQPtr retval = new MoodycamelBCQType;
+	if (retval == nullptr) {
+		return 0;
+	}
+	*handle = retval;
+	return 1;
+}
+
+int moodycamel_bcq_destroy(MoodycamelBCQHandle handle)
+{
+	delete reinterpret_cast<MoodycamelBCQPtr>(handle);
+	return 1;
+}
+
+int moodycamel_bcq_enqueue(MoodycamelBCQHandle handle, MoodycamelValue value)
+{
+	if (handle == nullptr) {
```

Comment:
if we're not checking this on `destroy` I think it's safe to remove this check here too and save a couple cycles.